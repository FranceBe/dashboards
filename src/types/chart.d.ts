export type ChartProps = {
  data: Record<string, string | number>[]
  valueKey: string
  nameKey: string
}
