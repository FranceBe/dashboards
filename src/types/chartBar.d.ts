import any = jasmine.any

export type ChartBarProps = {
  data: Record<string, string | number>[]
  valueKey: string
  nameKey: string
}
