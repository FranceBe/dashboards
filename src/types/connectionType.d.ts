export type ConnectionTypeProps = {
  connection_type: 'ethernet' | 'wifi' | 'cellular'
}
