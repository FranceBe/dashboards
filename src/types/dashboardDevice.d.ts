export type DashboardDeviceable = {
  connection_type: 'ethernet' | 'wifi' | 'cellular'
  last_seen_at: string
  mac_wifi: string
  serial_number: string
  sim_id: number
  status: 'disconnected' | 'connected'
  url: string
  voltage: number
}
