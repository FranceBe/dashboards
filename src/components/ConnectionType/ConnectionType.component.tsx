// ConnectionType component to display an icon matching
// The device connection type with a specific color
import Tooltip from '@material-ui/core/Tooltip'
import SettingsEthernetIcon from '@material-ui/icons/SettingsEthernet'
import SignalCellularAltIcon from '@material-ui/icons/SignalCellularAlt'
import WifiIcon from '@material-ui/icons/Wifi'
import { ConnectionTypeContainer } from 'components/ConnectionType/connectionType.style'
import React, { ReactElement } from 'react'
import { ConnectionTypeProps } from 'types/connectionType'

export const ConnectionType: React.FC<ConnectionTypeProps> = ({ connection_type }) => {
  const iconByType: Record<'ethernet' | 'wifi' | 'cellular', ReactElement> = {
    cellular: <SignalCellularAltIcon data-testid={'icon-cellular'} />,
    ethernet: <SettingsEthernetIcon data-testid={'icon-ethernet'} />,
    wifi: <WifiIcon data-testid={'icon-wifi'} />,
  }

  return (
    <ConnectionTypeContainer data-testid={'connection_type-container'} type={connection_type}>
      <Tooltip title={connection_type}>{iconByType[connection_type]}</Tooltip>
    </ConnectionTypeContainer>
  )
}

export default ConnectionType
