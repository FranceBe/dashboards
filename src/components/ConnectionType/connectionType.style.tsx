import React from 'react'
import styled from 'styled-components'
import { fontSizes, palette, spaces } from 'styles/variables'

export const ConnectionTypeContainer = styled((props) => <div {...props} />)<{
  type: 'ethernet' | 'wifi' | 'cellular'
}>`
  background-color: ${(props) => {
    if (props.type === 'wifi') {
      return palette.purple_primary
    } else if (props.type === 'ethernet') {
      return palette.pink_primary
    } else {
      return palette.yellow_light
    }
  }};
  width: fit-content;
  border-radius: ${spaces.small};
  padding: ${spaces.regular};
  font-size: ${fontSizes.small};
`
