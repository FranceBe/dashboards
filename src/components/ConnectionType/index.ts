import ConnectionTypeComponent, {
  ConnectionType,
} from 'components/ConnectionType/ConnectionType.component'

export { ConnectionType }

export default ConnectionTypeComponent
