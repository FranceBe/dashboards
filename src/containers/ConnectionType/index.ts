import ConnectionTypeContainer, {
  ConnectionType,
} from 'containers/ConnectionType/ConnectionType.container'

export { ConnectionType }

export default ConnectionTypeContainer
