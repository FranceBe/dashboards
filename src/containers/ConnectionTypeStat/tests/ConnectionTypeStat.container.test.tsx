import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'
import ConnectionType from 'containers/ConnectionTypeStat'
import React from 'react'
import { ConnectionTypeDataable } from 'types/connectionType'

describe('ConnectionType', () => {
  const devicesConnectionType: ConnectionTypeDataable[] = [
    { connection_type: 'ethernet', length: 400 },
    { connection_type: 'wifi', length: 300 },
    { connection_type: 'cellular', length: 300 },
  ]
  it('should match snapshot', () => {
    const { container } = render(<ConnectionType devicesConnectionType={devicesConnectionType} />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render a ChartPie component', () => {
    render(<ConnectionType devicesConnectionType={devicesConnectionType} />)
    expect(screen.getByTestId('chart-pie')).toBeInTheDocument()
  })
})
