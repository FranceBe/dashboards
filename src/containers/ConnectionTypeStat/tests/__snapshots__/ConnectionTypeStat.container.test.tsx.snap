// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionType should match snapshot 1`] = `
<div
  class="sc-dlnjwi gPscuh"
>
  <h2>
    Connection Type Ratio
  </h2>
  <div
    data-testid="chart-pie"
  >
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 400px; height: 400px;"
    >
      <svg
        class="recharts-surface"
        cx="50%"
        cy="50%"
        height="400"
        version="1.1"
        viewBox="0 0 400 400"
        width="400"
      >
        <defs>
          <clippath
            id="recharts1-clip"
          >
            <rect
              height="390"
              width="390"
              x="5"
              y="5"
            />
          </clippath>
        </defs>
        <g
          class="recharts-layer recharts-pie"
        >
          <g
            class="recharts-layer"
          >
            <g
              class="recharts-layer recharts-pie-sector"
            />
            <g
              class="recharts-layer recharts-pie-sector"
            />
            <g
              class="recharts-layer recharts-pie-sector"
            />
          </g>
        </g>
      </svg>
      <div
        class="recharts-tooltip-wrapper"
        style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="sc-gtsrHT dZVimF"
        >
          ethernet
          : 
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
